src/objects/line_manager/line_manager.gd

```diff
@@ -17,6 +21,21 @@ func _spawn_kid() -> void:
 	var kid := _kid_scene.instantiate() as Kid
 	add_sibling(kid)
 	kid.global_position = global_position
+	kid.enter_line.connect(_on_enter_line)
+	kid.bouncing.connect(_on_bouncing)
+
+func _game_over() -> void:
+	print("Game Over!!")
 
 func _on_timeout() -> void:
 	_spawn_kid()
+
+func _on_enter_line() -> void:
+	_line_count += 1
+	if _line_count >= _max_kids_inline:
```

Comment:
Maybe we can allow the max and only do game over if it exceeds. Using > instead of >=